### ISSUE TITLE
Docker build failed with error `COPY failed: missing build context` when base image have `ONBUILD COPY` command

### ISSUE THREAD

**Issue by donnykurnia** [docker/for-mac#1677]
### Expected behavior
docker-compose build successfully build the docker image

### Actual behavior

```
$ docker build . -t test-copy-onbuild -f Dockerfile-onbuild
Sending build context to Docker daemon  25.09kB
Step 1/3 : FROM ubuntu
 ---> ebcd9d4fca80
Step 2/3 : RUN mkdir -p /app/user
 ---> Using cache
 ---> b9456eee5f86
Step 3/3 : ONBUILD copy Gemfile Gemfile.lock /app/user/
 ---> Running in b73a90fd830f
 ---> aaed3f6e32ba
Removing intermediate container b73a90fd830f
Successfully built aaed3f6e32ba
Successfully tagged test-copy-onbuild:latest

$ docker build . -t test-copy
Sending build context to Docker daemon  25.09kB
Step 1/1 : FROM test-copy-onbuild
# Executing 1 build trigger...
Step 1/1 : COPY Gemfile Gemfile.lock /app/user/
COPY failed: missing build context
```

### Information

  - Full output of the diagnostics from "Diagnose & Feedback" in the menu

```
Docker for Mac: version: 17.06.0-rc1-ce-mac13 (4cdec4294a50b2233146b09469b49937dabdebdd)
macOS: version 10.11.6 (build: 15G1421)
logs: /tmp/26695482-50C0-4632-BF90-16EEBDECB94E/20170602-084030.tar.gz
[OK]     db.git
[OK]     vmnetd
[OK]     dns
[OK]     driver.amd64-linux
[OK]     virtualization VT-X
[OK]     app
[OK]     moby
[OK]     system
[OK]     moby-syslog
[OK]     db
[OK]     env
[OK]     virtualization kern.hv_support
[OK]     slirp
[OK]     osxfs
[OK]     moby-console
[OK]     logs
[OK]     docker-cli
[OK]     menubar
[OK]     disk
```

Dockerfile-onbuild

```
FROM ubuntu

RUN  mkdir -p /app/user
ONBUILD COPY ["Gemfile", "Gemfile.lock", "/app/user/"]
```

Dockerfile

```
FROM test-copy-onbuild
```

If the `Dockerfile` use `COPY` directly, it successed:

`Dockerfile-no-onbuild`:

```
FROM ubuntu

RUN  mkdir -p /app/user
COPY ["Gemfile", "Gemfile.lock", "/app/user/"]
```

```
$ docker build . -t test-copy-no-onbuild -f Dockerfile-no-onbuild
Sending build context to Docker daemon  26.11kB
Step 1/3 : FROM ubuntu
 ---> ebcd9d4fca80
Step 2/3 : RUN mkdir -p /app/user
 ---> Using cache
 ---> b9456eee5f86
Step 3/3 : COPY Gemfile Gemfile.lock /app/user/
 ---> Using cache
 ---> ea8339d9b1e4
Successfully built ea8339d9b1e4
Successfully tagged test-copy-no-onbuild:latest
```

### Steps to reproduce the behavior

  1. run either `docker-compose build [service name]` or `docker build .` using base image that have `ONBUILD COPY` command in that base image Dockerfile.
  2. Try the `Dockerfile` and `docker build` commands I've given above.

Update:
I try a simple Dockerfile building 2 image. Base image have `ONBUILD COPY` command and when building second image, it failed.

Additional information:
Before update to `17.06.0-rc1-ce-mac13`, this is working without error.

**Comment by errordeveloper** [comment]
I am also seeing this, and my script is even simpler, it's using a an official image.


```
FROM node:6-onbuild
EXPOSE 8080
```

**Comment by chrisfilo** [comment]
Same here.

**Comment by dnephin** [comment]
related issue moby/moby#33493

**Comment by techmaniack** [comment]
Same here.
I am using `ruby:2.3.0-onbuild`
and build fails with 
```
Step 1/1 : COPY Gemfile /usr/src/app/
COPY failed: missing build context
```

any quick fix for this one?

**Comment by donnykurnia** [comment]
Fixed in 17.06.0-ce-rc2
